--- gcolor2-0.4.orig/src/callbacks.c	2026-05-21 01:35:47.042541678 +0200
+++ gcolor2-0.4/src/callbacks.c	2026-05-21 01:35:47.046792016 +0200
@@ -9,6 +9,8 @@
 #include "interface.h"
 #include "support.h"
 
+#include <string.h>
+#include <glib/gprintf.h>
 
 GtkWidget        *savedialog;
 gchar            *colorname;
--- gcolor2-0.4.orig/src/callbacks.h	2026-05-21 01:35:47.042513447 +0200
+++ gcolor2-0.4/src/callbacks.h	2026-05-21 01:37:07.086219023 +0200
@@ -41,9 +41,9 @@
                            "\302\267 Usability / user interface enhancements\n" \
                            "\302\267 Code cleanup"
 
-GtkWidget *gcolor2;
-GtkWidget *menu;
-GdkColor   colorvalue;
+extern GtkWidget *gcolor2;
+extern GtkWidget *menu;
+extern GdkColor   colorvalue;
 
 enum
 {
@@ -55,6 +55,11 @@
 };
 
 gchar* get_user_file();
+gchar* get_system_file();
+
+void add_rgb_file (gchar *filename, gchar *type);
+
+void add_list_color (gchar *spec, gchar *name, gchar *type, gboolean is_new_color);
 
 void show_file_error (gchar* message);
 
--- gcolor2-0.4.orig/src/main.c	2026-05-21 01:35:47.042373544 +0200
+++ gcolor2-0.4/src/main.c	2026-05-21 01:37:18.907509060 +0200
@@ -8,11 +8,17 @@
 #include "interface.h"
 #include "support.h"
 #include "callbacks.h"
+#include <stdlib.h>
+#include <string.h>
+#include <glib/gprintf.h>
 
 GtkListStore *liststore;
 GdkWindow    *gdkwin;
 gchar        *user_filename;
 gchar        *sys_filename;
+GtkWidget    *gcolor2;
+GtkWidget    *menu;
+GdkColor      colorvalue;
 
 
 void set_user_file ()
@@ -48,8 +54,6 @@
 
 void add_list_color (gchar *spec, gchar *name, gchar *type, gboolean is_new_color)
 {
-	GdkPixmap   *pixmap;
-	GdkBitmap   *mask;
 	GdkPixbuf   *buf;
 	GtkTreeIter  iter;
 	
@@ -103,7 +107,7 @@
 	gchar *name;
 	gchar  buffer[512];
 	gchar  spec[8];
-	gint   r, g, b, lr, lg, lb;
+	gint   r, g, b, lr=0, lg=0, lb=0;
 	
 	fp = fopen (filename, "r");
 	if (!fp)
@@ -140,7 +144,6 @@
 	GtkTreeViewColumn *column;
 	GtkWidget         *treeview;
 	GtkTreeSelection  *select;
-	gchar             *buf;
 	
 #ifdef ENABLE_NLS
 	bindtextdomain (GETTEXT_PACKAGE, PACKAGE_LOCALE_DIR);
